vue-components/src/components/Dropdown.vue

```diff
@@ -0,0 +1,227 @@
+<template>
+	<div
+		:class="[ 'wikit', 'wikit-Dropdown' ]"
+		@keydown="triggerKeyDown"
+		@keyup="triggerKeyUp"
+	>
+		<label
+			class="wikit-Dropdown__label"
+			@click="$refs.select.focus()"
+		>{{ label }}</label>
+		<div
+			class="wikit-Dropdown__select"
+			:tabindex="!disabled && '0'"
+			:disabled="disabled"
+			@click="onClick"
+			@blur="showMenu = false"
+			ref="select"
+		>
+			<span
+				class="wikit-Dropdown__selectedOption"
+				v-if="value"
+			>{{ value.label }}</span>
+			<span
+				class="wikit-Dropdown__placeholder"
+				v-else
+			>{{ placeholder }}</span>
+			<svg
+				class="wikit-Dropdown__arrow"
+				fill="none"
+				viewBox="0 0 12 12"
+				xmlns="http://www.w3.org/2000/svg"
+			><path fill="currentColor" d="M6 10L12 3.94231L11 3L6 7.98077L1 3L5.2958e-07 3.94231L6 10Z" /></svg>
+		</div>
+		<OptionsMenu
+			class="wikit-Dropdown__menu"
+			:menu-items="menuItems"
+			:selected-item-index="selectedItemIndex"
+			v-show="showMenu"
+			@select="onSelect"
+			@esc="onEsc"
+			ref="menu"
+		/>
+	</div>
+</template>
+
+<script lang="ts">
+import Vue, { PropType, VueConstructor } from 'vue';
+import { MenuItem } from '@/components/MenuItem';
+import OptionsMenu from '@/components/OptionsMenu.vue';
+import isEqual from 'lodash.isequal';
+
+export default ( Vue as VueConstructor<Vue & { $refs: {
+	menu: InstanceType<typeof OptionsMenu>;
+	select: HTMLElement;
+}; }> ).extend( {
+	name: 'Dropdown',
+	data() {
+		return {
+			showMenu: false,
+		};
+	},
+	props: {
+		menuItems: {
+			type: Array as PropType<MenuItem[]>,
+			default: (): [] => [],
+		},
+		disabled: {
+			type: Boolean,
+			default: false,
+		},
+		label: {
+			type: String,
+			default: '',
+		},
+		/**
+		 * The selected menu item, can be of type `MenuItem` or `null`.
+		 *
+		 * The data usually comes from the consumer's `v-model` annotation on the Lookup component.
+		 */
+		value: {
+			type: Object,
+			default: null,
+		},
+		placeholder: {
+			type: String,
+			default: '',
+		},
+	},
+	computed: {
+		selectedItemIndex(): number {
+			if ( this.value === null || this.menuItems.length === 0 ) {
+				return -1;
+			}
+
+			return this.menuItems.findIndex(
+				( menuItem ) => { return isEqual( menuItem, this.value ); },
+				this,
+			);
+		},
+	},
+	methods: {
+		triggerKeyDown( event: KeyboardEvent ): void {
+			switch ( event.key ) {
+				case 'Enter':
+					if ( !this.showMenu ) {
+						this.showMenu = true;
+						return;
+					}
+					break;
```

Comment:
I think this causes an issue where the menu does open on `keydown` but immediately closes (since the `OptionsMenu` registers a `keyup` event an emits a `select`...). See screen capture below:

https://drive.google.com/file/d/1oszJ2HPv4sbDdk32_nrSwNAgmtd25SsT/view?usp=sharing

Reply:
Ah, got it. This is because we open the menu on key-down, but select something and close the menu on key-up. We probably should both on the same event :thinking: 

Do you or @SaiSan-WMDE have a preference for which event to use for the Enter-key behavior, Key-Down or Key-Up? I think I can also make it work for different events, but it will be more complex.

Reply:
> We probably should both on the same event thinking

No, I think the problem here is the scope of the event, I think maybe the `keydown` enter should only operate on the child div itself rather than the whole parent (not entirely positive, didn't try it out)

Reply:
I only see the issue happening whenever `enter` is pressed to open the menu after a selection has been made. Keyboard navigation seems to work fine until then?  The desired behavior in that case would be: the menu expands and the focus is placed on the currently selected option in the list.

While checking this I realized that another keyboard navigation behavior is missing in the menu:

- If down arrow is pressed when the last option in the menu is focused, the focus moves to the first menu item.
- If up arrow is pressed when the first option in the menu is focused, the focus moves to the last item of the menu. 

I wasn't sure of this behavior (some W3C components – like [this combobox](https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.1pattern/listbox-combo.html) – display it, some –like [this dropdown](https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html) –  don't) and ended up documenting it only because this is how OOUI menus work (accessibility was an important focus in that project).

Reply:
I think the fix for this issue is now in #302 as the root cause for this problem already exists in master, even though it is not apparent for the Lookup.